### ISSUE TITLE
Memory leak?

### ISSUE THREAD

**Issue by mikegagnon** [gnosek/nginx-upstream-fair#6]
In ngx_http_upstream_fair_module.c, beginning on line 604 

```
/* setup our wrapper around rr */
peers = ngx_palloc(cf->pool, sizeof *peers);
if (peers == NULL) {
    return NGX_ERROR;
}
peers = us->peer.data;
```

Looks like a bug to allocate memory to peers then immediately re-assign peers to something else. I'm not familiar with nginx code base, so I apologize if this is correct and I am misunderstanding something.

**Comment by gnosek** [comment]
Indeed, it looks dumb. I haven't looked at this code for a _very_ long time so I cannot tell you right now what it should be..

But still, the allocation is not a leak thanks to Nginx pools.

**Comment by fankeke** [comment]
indeed,I checked the code . 
so there is no need to do this:peers = ngx_palloc(cf->pool, sizeof *peers);
